runtime/appruntime/metrics/metrics.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"net/http"
+	"strings"
+
+	"encore.dev/beta/errs"
+)
+
+var r = strings.NewReplacer(
+	" ", "_",
+	"-", "_",
+	"'", "_",
+)
+
+type Manager struct {
+	exp Exporter
+}
+
+type Exporter interface {
+	IncCounter(name string, tags ...string)
+	Observe(name string, key string, value float64, tags ...string)
+}
+
+func NewManager(exp Exporter) *Manager {
+	return &Manager{exp: exp}
+}
+
+func (m *Manager) ReqEnd(service, endpoint string, err error, httpStatus int, durSecs float64) {
+	code := code(err, httpStatus)
+	m.exp.IncCounter(
+		"e_requests_total",
+		"service", service,
+		"endpoint", endpoint,
+		"code", code,
+	)
+	m.exp.Observe(
+		"e_request_duration_seconds",
+		"duration", durSecs,
+		"service", service,
+		"endpoint", endpoint,
+		"code", code,
+	)
+}
+
+func code(err error, httpStatus int) string {
+	if err != nil {
+		e := errs.Convert(err).(*errs.Error)
+		return e.Code.String()
+	}
+
+	code := http.StatusText(httpStatus)
```

Comment:
I don't think the output of `http.StatusText` is suitable for time series values; they're intended as human-readable text. What's more, most people will not know off the top of their head what HTTP status code `non_authoritative_information` refers to, which makes it difficult to reason about and search their code base for.

I think we should either do:

(a) only HTTP status codes (`400` etc), regardless of error code
(b) use our codes (`internal` etc) when the code matches one of those and `http_123` otherwise
(c) output both when we can (but might not work with AWS tag limits)

What do you think @DomBlack?

Reply:
I think using our own codes makes the most sense here; it aligns with how people think about their programs when writing errors in Encore

Reply:
So you're advocating for (b)? Or something else?

Reply:
(b)

Reply:
Sounds good to me too, I've pushed a commit that implements (b).